x/ecocredit/server/genesis.go

```diff
@@ -38,56 +38,37 @@ func (s serverImpl) InitGenesis(ctx types.Context, cdc codec.Codec, data json.Ra
 		return nil, err
 	}
 
-	var genesisState ecocredit.GenesisState
-	r, err := jsonSource.OpenReader(protoreflect.FullName(proto.MessageName(&genesisState)))
+	var params core.Params
+	r, err := jsonSource.OpenReader(protoreflect.FullName(proto.MessageName(&params)))
 	if err != nil {
 		return nil, err
 	}
 
 	if r == nil { // r is nil when theres no table data, so we can just unmarshal the data given
 		bz := bytes.NewBuffer(data)
-		err = (&jsonpb.Unmarshaler{AllowUnknownFields: true}).Unmarshal(bz, &genesisState)
+		err = (&jsonpb.Unmarshaler{AllowUnknownFields: true}).Unmarshal(bz, &params)
 		if err != nil {
 			return nil, err
 		}
 	} else { // r is not nil, so there is table data and we can just use r.
-		err = (&jsonpb.Unmarshaler{AllowUnknownFields: true}).Unmarshal(r, &genesisState)
+		err = (&jsonpb.Unmarshaler{AllowUnknownFields: true}).Unmarshal(r, &params)
 		if err != nil {
 			return nil, err
 		}
 	}
 
-	s.paramSpace.SetParamSet(ctx.Context, &genesisState.Params)
+	s.paramSpace.SetParamSet(ctx.Context, &params)
+	/*
+		store := ctx.KVStore(s.storeKey)
 
-	if err := s.creditTypeSeqTable.Import(ctx, genesisState.Sequences, 0); err != nil {
-		return nil, errors.Wrap(err, "sequences")
-	}
-
-	if err := s.classInfoTable.Import(ctx, genesisState.ClassInfo, 0); err != nil {
-		return nil, errors.Wrap(err, "class-info")
-	}
-
-	if err := s.projectInfoTable.Import(ctx, genesisState.ProjectInfo, 0); err != nil {
-		return nil, errors.Wrap(err, "project-info")
-	}
-
-	if err := s.projectInfoSeq.InitVal(ctx, genesisState.ProjectSeqNum); err != nil {
-		return nil, errors.Wrap(err, "project seq")
-	}
-
-	if err := s.batchInfoTable.Import(ctx, genesisState.BatchInfo, 0); err != nil {
-		return nil, errors.Wrap(err, "batch-info")
-	}
-
-	store := ctx.KVStore(s.storeKey)
-	if err := setBalanceAndSupply(store, genesisState.Balances); err != nil {
-		return nil, err
-	}
-
-	if err := validateSupplies(store, genesisState.Supplies); err != nil {
-		return nil, err
-	}
+			if err := setBalanceAndSupply(store, genesisState.Balances); err != nil {
+				return nil, err
+			}
 
+			if err := validateSupplies(store, genesisState.Supplies); err != nil {
+				return nil, err
+			}
+	*/
```

Comment:
I think we should focus on getting #977 merged first as there is some overlap here.

Reply:
sounds good to me 👍🏻